packages/recommend-react/src/Recommend.tsx

```diff
@@ -30,7 +30,7 @@ function isRegistered<TObject>(
   return Boolean(widgets.find((w) => dequal(w.param, param)));
 }
 
-function getCacheKey<TObject>(queries: Array<BatchQuery<TObject>>) {
+export function getCacheKey<TObject>(queries: Array<BatchQuery<TObject>>) {
```

Comment:
All these exports mean that users will be able to use these functions, which increases our API surface. We want to avoid that. Could be easily done by moving these functions in their own file.

Reply:
Good catch. Moved these functions into a dedicated helper file. 👍